packages/mobile/src/screens/favorites-screen/TracksTab.tsx

```diff
@@ -93,13 +129,55 @@ export const TracksTab = () => {
     isReachable
   ])
 
-  useFocusEffect(handleFetchSaves)
+  const handleMoreFetchSaves = useCallback(() => {
+    if (
+      allTracksFetched ||
+      isLoading ||
+      isFetchingMore ||
+      (isOfflineModeEnabled && !isReachable) ||
+      tracks.length < fetchPage * FETCH_LIMIT
+    ) {
+      return
+    }
 
-  const [filterValue, setFilterValue] = useState('')
-  const isPlaying = useSelector(getPlaying)
-  const playingUid = useSelector(getUid)
-  const savedTracksStatus = useSelector(getSavedTracksStatus)
-  const savedTracks = useProxySelector(getTracks, [])
+    const nextPage = fetchPage + 1
+    dispatch(
+      fetchMoreSaves(filterValue, '', '', nextPage * FETCH_LIMIT, FETCH_LIMIT)
+    )
+    setFetchPage(nextPage)
+  }, [
+    allTracksFetched,
+    dispatch,
+    fetchPage,
+    filterValue,
+    isFetchingMore,
+    isLoading,
+    isOfflineModeEnabled,
+    isReachable,
+    tracks.length
+  ])
+
+  useEffect(() => {
+    const allTracks = tracks.every((track) => track.kind === Kind.TRACKS)
```

Comment:
This does feel like it would be expensive as the list gets long? What do you think?

Reply:
simple equality check so doesn't seem too bad. I think that it may run more often than it needs to, but would wait to see if we have perf issues before trying to fix this